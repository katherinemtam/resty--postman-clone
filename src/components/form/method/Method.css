.Method input[type="radio"] {
	 visibility: hidden;
	 height: 0;
	 width: 0;
	 margin-left: 0;
}

.Method label {
  font-family: Verdana, Geneva, Tahoma, sans-serif;
	padding: 4px 16px;
	cursor: pointer;
}

.Method input[type="radio"]:checked + label {
	 background-color: #204060;
	 color: white;
}
