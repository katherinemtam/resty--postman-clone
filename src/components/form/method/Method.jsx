import React from 'react';
import PropTypes from 'prop-types';
import styles from './Method.css';

const Method = ({ method, onChange }) => (
  <div className={styles.Method}>
    <label htmlFor="get">GET</label>
    <input
      id="get"
      type="radio"
      name="method"
      value="GET"
      onChange={onChange}
      checked={method === 'GET'}
    />
    <label htmlFor="post">POST</label>
    <input
      id="post"
      type="radio"
      name="method"
      value="POST"
      onChange={onChange}
      checked={method === 'POST'}
    />
    <label htmlFor="put">PUT</label>
    <input
      id="put"
      type="radio"
      name="method"
      value="PUT"
      onChange={onChange}
      checked={method === 'PUT'}
    />
    <label htmlFor="patch">PATCH</label>
    <input
      id="patch"
      type="radio"
      name="method"
      value="PATCH"
      onChange={onChange}
      checked={method === 'PATCH'}
    />
    <label htmlFor="delete">DELETE</label>
    <input
      id="delete"
      type="radio"
      name="method"
      value="DELETE"
      onChange={onChange}
      checked={method === 'DELETE'}
    />
  </div>
);

Method.propTypes = {
  method: PropTypes.string,
  onChange: PropTypes.func,
};

export default Method;


