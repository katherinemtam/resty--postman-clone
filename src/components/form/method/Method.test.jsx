import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Method from './Method';

describe('Method component', () => {
  afterEach(() => cleanup());
  it('renders Method', () => {
    const { asFragment } = render(<Method />);
    expect(asFragment()).toMatchSnapshot();
  });
});
