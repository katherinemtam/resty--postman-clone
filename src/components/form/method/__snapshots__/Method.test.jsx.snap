// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Method component renders Method 1`] = `
<DocumentFragment>
  <div
    class="Method"
  >
    <div>
      <input
        id="POST"
        name="method"
        type="radio"
        value="POST"
      />
      <label
        for="POST"
      >
        POST
      </label>
    </div>
    <div>
      <input
        id="GET"
        name="method"
        type="radio"
        value="GET"
      />
      <label
        for="GET"
      >
        GET
      </label>
    </div>
    <div>
      <input
        id="PUT"
        name="method"
        type="radio"
        value="PUT"
      />
      <label
        for="PUT"
      >
        PUT
      </label>
    </div>
    <div>
      <input
        id="PATCH"
        name="method"
        type="radio"
        value="PATCH"
      />
      <label
        for="PATCH"
      >
        PATCH
      </label>
    </div>
    <div>
      <input
        id="DELETE"
        name="method"
        type="radio"
        value="DELETE"
      />
      <label
        for="DELETE"
      >
        DELETE
      </label>
    </div>
  </div>
</DocumentFragment>
`;
